WordPress/Classes/ViewRelated/Comments/Content/CommentContentHeaderView.swift

```diff
@@ -0,0 +1,197 @@
+        }
+    }
+
+    struct Configuration {
+        let avatarURL: URL?
+        let username: String
+        let handleAndTimestamp: String
+        let menu: [[MenuItem]]
+    }
+
+    typealias MenuItem = CommentContentHeaderMenu.Item
+}
+
+extension CommentContentHeaderView.Configuration {
+
+     init(avatarURL: URL?, username: String, handleAndTimestamp: String, menu: [CommentContentHeaderView.MenuItem]) {
+        self.init(
+            avatarURL: avatarURL,
+            username: username,
+            handleAndTimestamp: handleAndTimestamp,
+            menu: [menu]
+        )
+    }
+}
+
+struct CommentContentHeaderMenu: View {
+   let menu: [[Item]]
+
+   var body: some View {
+       Menu {
+           ForEach(Array(menu.enumerated()), id: \.offset) { sectionIndex, section in
+               if !section.isEmpty {
+                   Section {
+                       ForEach(Array(section.enumerated()), id: \.offset) { itemIndex, menuItem in
+                           button(for: menuItem)
+                       }
+                   }
+                   if sectionIndex < menu.count - 1 && !menu[sectionIndex + 1].isEmpty {
+                       Divider()
+                   }
+               }
+           }
+       } label: {
+           Image.DS.icon(named: .ellipsisHorizontal)
+               .imageScale(.small)
+               .frame(width: .DS.Padding.medium, height: .DS.Padding.medium)
+               .foregroundStyle(Color.DS.Foreground.secondary)
+       }
+   }
+
+   @ViewBuilder
+   func button(for menuItem: Item) -> some View {
+       switch menuItem {
+       case .userInfo(let action):
+           Button(action: action) {
+               Label {
+                   Text(Strings.userInfo)
+               } icon: {
+                   Image.DS.icon(named: .avatar)
+               }
+           }
+       case .share(let action):
+           Button(action: action) {
+               Label {
+                   Text(Strings.share)
+               } icon: {
+                   Image.DS.icon(named: .blockShare)
+               }
+           }
+       case .editComment(let action):
+           Button(action: action) {
+               Label {
+                   Text(Strings.editComment)
+               } icon: {
+                   Image.DS.icon(named: .edit)
+               }
+           }
+       case .changeStatus(let action):
+           Menu(Strings.changeStatus) {
+               Button(Strings.approve, action: { action(.approve) })
+               Button(Strings.pending, action: { action(.pending) })
+               Button(Strings.trash, action: { action(.trash) })
+               Button(Strings.spam, action: { action(.spam) })
+           }
+       }
+   }
+
+   enum Item {
+       case userInfo(() -> Void)
+       case share(() -> Void)
+       case editComment(() -> Void)
+       case changeStatus((Status) -> Void)
+
+       enum Status {
+           case approve, trash, spam, pending
+       }
+   }
```

Comment:
Wouldn't a ViewModel be more scalable here? The view can report to actions there. Instead of dealing with closures, we the VM could implement all the cases. For 1 or 2 events I think closures are fine but this to me feels a bit rigid in terms of scalability. Wdyt?

Reply:
I've tried having only 1 closure to report menu events, something like this:

```swift
  enum Item {
       case userInfo
       case share
       case editComment
       case changeStatus(Status)

       enum Status {
           case approve, trash, spam, pending
       }
   }
```

Then I wanted to have a closure with this signature that callers will implement

```swift
let onItemSelected: (Item) -> Void = { item in 
    ...
}
```

The challenge arises with the .changeStatus(Status) case in our Item enum. When configuring the menu:

```swift
let menu = [.userInfo, .share, .editComment, .changeStatus(???)]
```

It's unclear how to specify a status for `.changeStatus` statically, as it represents multiple potential actions (approve, trash, spam, pending).

Reply:
Can you share a pseudo-code of how the VM approach would work?